### DISCUSSION TITLE
There should be an Task.Run(Action<CancellationToken>, CancellationToken) overload

### DISCUSSION THREAD

**Discussion by soraphis** [dotnet/runtime#63575] [Ideas]
assume having a cancellable task:
```Csharp
    var cancellationToken = // ... 
    void taskAction(){
        // ....
        cancellationToken.ThrowIfCancellationRequested();
        // ...
    }

    Task.Run(taskAction, cancellationToken);
```

- taskAction will capture "cancellationToken"

the cancellationToken that is passed to Run could be passed from there as an argument to the taskAction, passing it by parameter instead of capturing it.

**Comment by huoyaoyuan** [comment]
The cancellation token in `Task.Run` is currently used by task scheduler and continuation. Not sure the suggestion for task body. @dotnet/area-system-threading-tasks

**Reply by soraphis** [reply]
the cancellationToken in Task.Run will only prevent the task from running at all if the token is already canceled before the task is scheduled.
which is super nice!

but assume you have a task with e.g. 2 long going methods in it, and you wanna check between those calls if the cancellationToken is already canceled (so that you save the second long going execution) you'd have to...

... a) pass the cancelation token into the method. which captures the cancellationToken - which would be nice to circumvent.
... b) split your task into two tasks with continuation passing the result. which might not be feasible, think of short-circuiting a loop - first action in your loop would be: `cancellationToken.ThrowIfCancellationRequested();` this is not easily done with continuation tasks

**Comment by Clockwork-Muse** [comment]
@huoyaoyuan - Probably that the `CancellationToken` is automatically passed to the task.

@soraphis - Current workaround:

```csharp
public static Task Run(Action<CancellationToken> action, CancellationToken token)
{
    // I believe this will be hoisted so that there will be no per-invocation Action instance.
    void RunTuple(object? t) 
    {
        // All because we have to unbox the token, alas.
        var (inputAction, inputToken) = (Tuple<Action<CancellationToken>, CancellationToken>)t;
        inputAction(inputToken);
    };
    return Task.Factory.StartNew(RunTuple, (action, token), token, TaskCreationOptions.DenyChildAttach, TaskScheduler.Default);
}
```

**Reply by soraphis** [reply]
return Task.Factory.StartNew(RunTuple, (action, token), token, TaskCreationOptions.DenyChildAttach, TaskScheduler.Default);

`(action, token)` is a ValueTuple which is assigned to an `object` parameter, causing boxing.

to circumvent this, their would be a StartNew method needed with generic parameter like:

        public Task StartNew<TState>(Action<TState> action, TState state);
